x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -951,6 +1089,23 @@ private static String getTokenDocumentId(String id) {
         return "token_" + id;
     }
 
+    private static String getTokenIdFromDocumentId(String docId) {
+        if (docId.substring(0, "token_".length()).equals("token_") == false) {
```

Comment:
Why not `docId.startsWith("token_")` ?

Reply:
Some early on PR review ( I think it was @jaymode ) commended `substring` over `startsWith` and it kind of stuck with me. I couldn't find it now and I may have taken it out of specific context and generalized the suggestion. I agree that `startsWith` reads easier though, I'll change this

Reply:
If there's a reason, then I'm happy with it, but I would expect `startsWith` to be more performant and more readable.